modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java

```diff
@@ -78,6 +86,15 @@ public void setUp() throws Exception {
 				RandomTestUtil.randomLocaleStringMap(), 2, 2, "HALF_EVEN",
 				false, RandomTestUtil.nextDouble(), true);
 
+		_commerceChannel = CommerceTestUtil.addCommerceChannel(
+			testGroup.getGroupId(), commerceCurrency.getCode());
+
+		_commerceChannel.setExternalReferenceCode(
+			StringUtil.toLowerCase(RandomTestUtil.randomString()));
+
+		_commerceChannel = _commerceChannelLocalService.updateCommerceChannel(
+			_commerceChannel);
+
```

Comment:
I think this is not needed because the external reference code is already autogenerated random if not provided